### PR TITLE
UIIN-2327: Disable deletion of a bound-with title

### PR DESCRIPTION
**Pull request by maccabeelevine**
## Purpose

According to the bound-with data model, an item should be bound-with linked
(via the bound-with parts API) to its own directly linked holdings record,
in addition to the one or more other holdings records it is bound with. 
Therefore it's incorrect to delete that directly-linked holdings record's 
relationship with the item record.

- Note that the backend API used for edits to the bound-with parts already enforces this relationship.  This PR only reflects that logic in the UI.
- As a corollary, note that even though the directly-linked bound-with title can't be removed via the UI, it will *automatically* be removed via that API call if it is the only remaining bound-with relationship on this item record.

## Approach
Disable the 'delete' action for a bound-with title on the Item Edit screen,
if that title (the indicated holdings record) is also directly linked to the item
as item.holdingsRecordId.

I enhanced FieldRow (used by RepeatableField) to support a function
to enable/disable the 'remove' action selectively per row, rather than 
universally.  For backwards compatibility it still supports and defaults
to expecting a boolean value.  I don't **think** this constitutes
a breaking change.

I made those changes to the ui-inventory version of these components, 
not the stripes-components version.  There are other UI differences between
 the identically-named components, and we wanted the UX to remain 
consistent between all of the ItemForm uses of RepeatableField.  Those 
differences include the layout of the delete button relative to the rest of 
the line, as well as the repetition of the field headers above each line.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [X] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

## Issues
https://issues.folio.org/browse/UIIN-2327

### CODE REVIEW

**Comment by maccabeelevine** [comment]
@MikeTaylor I would appreciate an early review and in particular some help with the test case(s).  I wrote up what I think is a reasonable pair of tests, but it fails due to no such elements being found.  I encounter this a lot and @mkuklis always manages to find the race condition and add some more `await` style logic so it works.  I tried to do that myself but no luck.  I think Michal is on vacation this week and @CharlotteWhitt cares about this functionality, so it may or may not be fine to wait for him.  Thanks!

**Comment by MikeTaylor** [comment]
Having discussed this with @maccabeelevine, we agreed that I'm not really competent to review this work. We may have to wait for @mkuklis.

**Comment by maccabeelevine** [comment]
No worries @MikeTaylor, I ended up creating a new file entirely but got it to work.

File: src/components/RepeatableField/FieldRow.js [hunk 1]
```diff
@@ -204,7 +204,10 @@ class FieldRow extends React.Component {
                         buttonStyle="link"
                         onClick={() => { this.handleRemove(fieldIndex, f); }}
                         aria-label={ariaLabel}
-                        disabled={!canDelete}
+                        disabled={
```
Comment (mkuklis) [inline src/components/RepeatableField/FieldRow.js:207]: @maccabeelevine should we also protect for the cases when `canDelete` is not passed at all?

Comment (maccabeelevine) [reply src/components/RepeatableField/FieldRow.js:207]: @mkuklis I think that's already taken care of by the defaultProps?

```
FieldRow.defaultProps = {
  canAdd: true,
  canEdit: true,
  canDelete: true,
};
```

Comment (mkuklis) [reply src/components/RepeatableField/FieldRow.js:207]: Ah I missed the default props! This all looks good.

**Review by mkuklis — APPROVED** [review]

**Pull request merged**